### ISSUE TITLE
Can't Install

### ISSUE THREAD

**Issue by DavidFerreira1002** [IASRobolab/MMT#1]
When I try to install it gives me this:

running install
running bdist_egg
running egg_info
writing MMT.egg-info/PKG-INFO
writing dependency_links to MMT.egg-info/dependency_links.txt
writing requirements to MMT.egg-info/requires.txt
writing top-level names to MMT.egg-info/top_level.txt
reading manifest file 'MMT.egg-info/SOURCES.txt'
writing manifest file 'MMT.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/mmt
copying build/lib/mmt/trainers.py -> build/bdist.linux-x86_64/egg/mmt
creating build/bdist.linux-x86_64/egg/mmt/utils
copying build/lib/mmt/utils/rerank.py -> build/bdist.linux-x86_64/egg/mmt/utils
copying build/lib/mmt/utils/__init__.py -> build/bdist.linux-x86_64/egg/mmt/utils
copying build/lib/mmt/utils/meters.py -> build/bdist.linux-x86_64/egg/mmt/utils
copying build/lib/mmt/utils/osutils.py -> build/bdist.linux-x86_64/egg/mmt/utils
copying build/lib/mmt/utils/serialization.py -> build/bdist.linux-x86_64/egg/mmt/utils
creating build/bdist.linux-x86_64/egg/mmt/utils/data
copying build/lib/mmt/utils/data/sampler.py -> build/bdist.linux-x86_64/egg/mmt/utils/data
copying build/lib/mmt/utils/data/preprocessor.py -> build/bdist.linux-x86_64/egg/mmt/utils/data
copying build/lib/mmt/utils/data/__init__.py -> build/bdist.linux-x86_64/egg/mmt/utils/data
copying build/lib/mmt/utils/data/transforms.py -> build/bdist.linux-x86_64/egg/mmt/utils/data
copying build/lib/mmt/utils/data/base_dataset.py -> build/bdist.linux-x86_64/egg/mmt/utils/data
copying build/lib/mmt/utils/logging.py -> build/bdist.linux-x86_64/egg/mmt/utils
copying build/lib/mmt/utils/lr_scheduler.py -> build/bdist.linux-x86_64/egg/mmt/utils
creating build/bdist.linux-x86_64/egg/mmt/evaluation_metrics
copying build/lib/mmt/evaluation_metrics/ranking.py -> build/bdist.linux-x86_64/egg/mmt/evaluation_metrics
copying build/lib/mmt/evaluation_metrics/__init__.py -> build/bdist.linux-x86_64/egg/mmt/evaluation_metrics
copying build/lib/mmt/evaluation_metrics/classification.py -> build/bdist.linux-x86_64/egg/mmt/evaluation_metrics
creating build/bdist.linux-x86_64/egg/mmt/metric_learning
copying build/lib/mmt/metric_learning/__init__.py -> build/bdist.linux-x86_64/egg/mmt/metric_learning
copying build/lib/mmt/metric_learning/kissme.py -> build/bdist.linux-x86_64/egg/mmt/metric_learning
copying build/lib/mmt/metric_learning/euclidean.py -> build/bdist.linux-x86_64/egg/mmt/metric_learning
copying build/lib/mmt/metric_learning/distance.py -> build/bdist.linux-x86_64/egg/mmt/metric_learning
creating build/bdist.linux-x86_64/egg/mmt/datasets
copying build/lib/mmt/datasets/msmt17.py -> build/bdist.linux-x86_64/egg/mmt/datasets
copying build/lib/mmt/datasets/__init__.py -> build/bdist.linux-x86_64/egg/mmt/datasets
copying build/lib/mmt/datasets/dukemtmc.py -> build/bdist.linux-x86_64/egg/mmt/datasets
copying build/lib/mmt/datasets/market1501.py -> build/bdist.linux-x86_64/egg/mmt/datasets
copying build/lib/mmt/__init__.py -> build/bdist.linux-x86_64/egg/mmt
copying build/lib/mmt/evaluators.py -> build/bdist.linux-x86_64/egg/mmt
creating build/bdist.linux-x86_64/egg/mmt/models
copying build/lib/mmt/models/resnet_ibn_a.py -> build/bdist.linux-x86_64/egg/mmt/models
copying build/lib/mmt/models/__init__.py -> build/bdist.linux-x86_64/egg/mmt/models
copying build/lib/mmt/models/resnet.py -> build/bdist.linux-x86_64/egg/mmt/models
copying build/lib/mmt/models/resnet_ibn.py -> build/bdist.linux-x86_64/egg/mmt/models
creating build/bdist.linux-x86_64/egg/mmt/loss
copying build/lib/mmt/loss/crossentropy.py -> build/bdist.linux-x86_64/egg/mmt/loss
copying build/lib/mmt/loss/triplet.py -> build/bdist.linux-x86_64/egg/mmt/loss
copying build/lib/mmt/loss/__init__.py -> build/bdist.linux-x86_64/egg/mmt/loss
copying build/lib/mmt/dist_metric.py -> build/bdist.linux-x86_64/egg/mmt
creating build/bdist.linux-x86_64/egg/mmt/feature_extraction
copying build/lib/mmt/feature_extraction/__init__.py -> build/bdist.linux-x86_64/egg/mmt/feature_extraction
copying build/lib/mmt/feature_extraction/database.py -> build/bdist.linux-x86_64/egg/mmt/feature_extraction
copying build/lib/mmt/feature_extraction/cnn.py -> build/bdist.linux-x86_64/egg/mmt/feature_extraction
byte-compiling build/bdist.linux-x86_64/egg/mmt/trainers.py to trainers.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/rerank.py to rerank.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/__init__.py to __init__.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/meters.py to meters.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/osutils.py to osutils.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/serialization.py to serialization.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/data/sampler.py to sampler.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/data/preprocessor.py to preprocessor.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/data/__init__.py to __init__.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/data/transforms.py to transforms.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/data/base_dataset.py to base_dataset.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/logging.py to logging.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/utils/lr_scheduler.py to lr_scheduler.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/evaluation_metrics/ranking.py to ranking.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/evaluation_metrics/__init__.py to __init__.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/evaluation_metrics/classification.py to classification.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/metric_learning/__init__.py to __init__.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/metric_learning/kissme.py to kissme.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/metric_learning/euclidean.py to euclidean.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/metric_learning/distance.py to distance.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/datasets/msmt17.py to msmt17.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/datasets/__init__.py to __init__.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/datasets/dukemtmc.py to dukemtmc.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/datasets/market1501.py to market1501.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/__init__.py to __init__.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/evaluators.py to evaluators.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/models/resnet_ibn_a.py to resnet_ibn_a.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/models/__init__.py to __init__.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/models/resnet.py to resnet.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/models/resnet_ibn.py to resnet_ibn.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/loss/crossentropy.py to crossentropy.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/loss/triplet.py to triplet.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/loss/__init__.py to __init__.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/dist_metric.py to dist_metric.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/feature_extraction/__init__.py to __init__.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/feature_extraction/database.py to database.cpython-38.pyc
byte-compiling build/bdist.linux-x86_64/egg/mmt/feature_extraction/cnn.py to cnn.cpython-38.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying MMT.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MMT.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MMT.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MMT.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MMT.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating 'dist/MMT-1.1.0-py3.8.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing MMT-1.1.0-py3.8.egg
Removing /usr/local/lib/python3.8/dist-packages/MMT-1.1.0-py3.8.egg
Copying MMT-1.1.0-py3.8.egg to /usr/local/lib/python3.8/dist-packages
MMT 1.1.0 is already the active version in easy-install.pth

Installed /usr/local/lib/python3.8/dist-packages/MMT-1.1.0-py3.8.egg
Processing dependencies for MMT==1.1.0
Searching for torch==1.1.0
Reading https://pypi.org/simple/torch/
No local packages or working download links found for torch==1.1.0
error: Could not find suitable distribution for Requirement.parse('torch==1.1.0')

my installed packages on a python venv are:

Package                       Version       
----------------------------- --------------
absl-py                       2.0.0         
actionlib                     1.14.0        
ai-utils                      1.0.0         
angles                        1.9.13        
attrs                         23.1.0        
base-local-planner            1.17.3        
bondpy                        1.8.6         
camera-calibration            1.17.0        
camera-calibration-parsers    1.12.0        
catkin                        0.8.10        
cffi                          1.16.0        
contourpy                     1.1.1         
controller-manager            0.20.0        
controller-manager-msgs       0.20.0        
cv-bridge                     1.16.2        
cycler                        0.12.1        
Cython                        3.0.5         
diagnostic-analysis           1.11.0        
diagnostic-common-diagnostics 1.11.0        
diagnostic-updater            1.11.0        
dynamic-reconfigure           1.7.3         
flatbuffers                   23.5.26       
fonttools                     4.45.0        
gazebo-plugins                2.9.2         
gazebo-ros                    2.9.2         
gencpp                        0.7.0         
geneus                        3.0.0         
genlisp                       0.4.18        
genmsg                        0.6.0         
gennodejs                     2.0.2         
genpy                         0.6.15        
image-geometry                1.16.2        
importlib-resources           6.1.1         
interactive-markers           1.12.0        
joblib                        1.3.2         
joint-state-publisher         1.15.1        
joint-state-publisher-gui     1.15.1        
kiwisolver                    1.4.5         
laser-geometry                1.6.7         
matplotlib                    3.7.4         
mediapipe                     0.10.8        
message-filters               1.16.0        
numpy                         1.24.4        
opencv-contrib-python         4.8.1.78      
opencv-python                 4.8.1.78      
packaging                     23.2          
Pillow                        10.1.0        
pip                           20.0.2        
pkg-resources                 0.0.0         
protobuf                      3.20.3        
pycocotools                   2.0.7         
pycparser                     2.21          
pyparsing                     3.1.1         
python-dateutil               2.8.2         
python-qt-binding             0.4.4         
qt-dotgraph                   0.4.2         
qt-gui                        0.4.2         
qt-gui-cpp                    0.4.2         
qt-gui-py-common              0.4.2         
resource-retriever            1.12.7        
rosbag                        1.16.0        
rosboost-cfg                  1.15.8        
rosclean                      1.15.8        
roscreate                     1.15.8        
rosgraph                      1.16.0        
roslaunch                     1.16.0        
roslib                        1.15.8        
roslint                       0.12.0        
roslz4                        1.16.0        
rosmake                       1.15.8        
rosmaster                     1.16.0        
rosmsg                        1.16.0        
rosnode                       1.16.0        
rosparam                      1.16.0        
rospy                         1.16.0        
rosserial-python              0.9.2         
rosservice                    1.16.0        
rostest                       1.16.0        
rostopic                      1.16.0        
rosunit                       1.15.8        
roswtf                        1.16.0        
rqt-action                    0.4.9         
rqt-bag                       0.5.1         
rqt-bag-plugins               0.5.1         
rqt-console                   0.4.12        
rqt-dep                       0.4.12        
rqt-graph                     0.4.14        
rqt-gui                       0.5.3         
rqt-gui-py                    0.5.3         
rqt-image-view                0.4.17        
rqt-launch                    0.4.9         
rqt-logger-level              0.4.12        
rqt-moveit                    0.5.11        
rqt-msg                       0.4.10        
rqt-nav-view                  0.5.7         
rqt-plot                      0.4.13        
rqt-pose-view                 0.5.11        
rqt-publisher                 0.4.10        
rqt-py-common                 0.5.3         
rqt-py-console                0.4.10        
rqt-reconfigure               0.5.5         
rqt-robot-dashboard           0.5.8         
rqt-robot-monitor             0.5.15        
rqt-robot-steering            0.5.12        
rqt-runtime-monitor           0.5.10        
rqt-rviz                      0.7.0         
rqt-service-caller            0.4.10        
rqt-shell                     0.4.11        
rqt-srv                       0.4.9         
rqt-tf-tree                   0.6.4         
rqt-top                       0.4.10        
rqt-topic                     0.4.13        
rqt-web                       0.4.10        
rviz                          1.14.20       
scikit-learn                  1.3.2         
scipy                         1.10.1        
sensor-msgs                   1.13.1        
setuptools                    44.0.0        
six                           1.16.0        
smach                         2.5.2         
smach-ros                     2.5.2         
smclib                        1.8.6         
sounddevice                   0.4.6         
tf                            1.13.2        
tf-conversions                1.13.2        
tf2-geometry-msgs             0.7.7         
tf2-kdl                       0.7.7         
tf2-py                        0.7.7         
tf2-ros                       0.7.7         
threadpoolctl                 3.2.0         
topic-tools                   1.16.0        
torch                         1.10.2+cu113  
torchaudio                    0.10.2+rocm4.1
torchvision                   0.11.3+cu113  
turtlebot3-example            1.2.5         
turtlebot3-teleop             1.2.5         
typing-extensions             4.8.0         
wheel                         0.41.3        
xacro                         1.14.16       
zipp                          3.17.0 

I have torch above 1.1.0, but I can't seem to install that specific version.

**Comment by FedericoRollo** [comment]
Move to the MMT directory in the terminal and try:
```python
pip install -e .
```

In the future please assign the issues to me when creating new ones.

**Comment by DavidFerreira1002** [comment]
Got this error:

ERROR: Could not find a version that satisfies the requirement torch==1.1.0 (from MMT==1.1.0) (from versions: 1.4.0, 1.5.0, 1.5.1, 1.6.0, 1.7.0, 1.7.1, 1.8.0, 1.8.1, 1.9.0, 1.9.1, 1.10.0, 1.10.1, 1.10.2, 1.11.0, 1.12.0, 1.12.1, 1.13.0, 1.13.1, 2.0.0, 2.0.1, 2.1.0, 2.1.1)
ERROR: No matching distribution found for torch==1.1.0 (from MMT==1.1.0)

**Comment by FedericoRollo** [comment]
that's strange. there are no dependencies from torch 1.1.0 in the installer. 

try to:
1. check that you have the right CUDA version installed (11.3). see CUDA website if needed.
2. uninstall all torch versions and reinstall following the procedure of ai_utils
3. maybe check also what python version you are using.

tomorrow I'll check in detail but this is a strange behaviour. I have recently installed it on another computer and it worked.

**Comment by FedericoRollo** [comment]
if the problem persists you can try to delete the torch>=1.4.0 and torchvision>=0.5.0 dependencies from setup.py file in the MMT directory.

**Comment by DavidFerreira1002** [comment]
It has installed now, I deleted the dependencies for torch and torchvision.
